datashape/discovery.py

```diff
@@ -396,3 +396,29 @@ def descendents(d, x):
         children -= desc
         desc.update(children)
     return desc
+
+
+def _tp_eq(a, b):
+    return a == b or b is null
```

Comment:
i'd probably just inline this code with a short comment explaining what it means